### PR TITLE
#863 - Allow opening the create api token dialog on the /settings/api page via a URL

### PR DESCRIPTION
**Pull request by rupin90**
Commit one - just includes to open the Generate API key modal.

Commit two - includes changes for prepopulation. 

Let me know any feedback.

### CODE REVIEW

**Review by sywhb — COMMENTED** [review]
Feel free to modify the `FormModal` and the other form component.
I'm still getting familiar with React

**Comment by jacksonh** [comment]
Looks like one small bug, if you go to the page without any params, and tap the `Create` button, you will get `undefined` in the name text input:

Screen Shot 2022-06-27 at 22 51 03

File: packages/web/styles/globals.css [hunk 1]
```diff
@@ -274,6 +274,15 @@ div#appleid-signin {
...
 
+select {
+  color: var(--colors-grayTextContrast);
+  background-color: transparent;
+}
+
+select option {
+  background-color: transparent;
+}
+
```
Comment (rupin90) [inline packages/web/styles/globals.css:285]: Good to know, Thank you @jacksonh

**Comment by jacksonh** [comment]
Hey @rupin90 one requirement i forgot to add on this. Can we also pass in `expire=` and allow the options `7, 30, 90, 365, never`? This would set the initial value in the `Expires In` drop down. 

I will implement support for support for `never` in the API today.

**Comment by rupin90** [comment]
> Hey @rupin90 one requirement i forgot to add on this. Can we also pass in `expire=` and allow the options `7, 30, 90, 365, never`? This would set the initial value in the `Expires In` drop down.
> 
> I will implement support for support for `never` in the API today.

Yes sure sounds good.

**Comment by rupin90** [comment]
Changes pushed and then resolved any conflicts. 
This is ready for the Review.

@jacksonh @satindar

File: packages/web/pages/settings/api.tsx [hunk 2]
```diff
@@ -72,6 +98,53 @@ export default function Api(): JSX.Element {
     revalidate()
   }
 
+  function onAdd() {
+    return setFormInputs([
```
Comment (sywhb) [inline packages/web/pages/settings/api.tsx:102]: we might want to wrap these default settings in another function

**Comment by jacksonh** [comment]
Closing this one out. Lets just plan in future to create a page for this, similar to the Readwise integration page.

**Comment by rupin90** [comment]
Okay sounds good!!

**Pull request closed without merging**